src/main/java/com/gotocompany/depot/maxcompute/client/insert/session/StreamingSessionManager.java

```diff
@@ -0,0 +1,58 @@
+package com.gotocompany.depot.maxcompute.client.insert.session;
+
+import com.aliyun.odps.tunnel.TableTunnel;
+
+import com.aliyun.odps.tunnel.TunnelException;
+import com.google.common.cache.CacheBuilder;
+import com.google.common.cache.CacheLoader;
+import com.google.common.cache.LoadingCache;
+import com.gotocompany.depot.config.MaxComputeSinkConfig;
+
+public final class StreamingSessionManager {
+
+    private final LoadingCache<String, TableTunnel.StreamUploadSession> sessionCache;
```

Comment:
What is the context behind using this LoadingCache?

Reply:
as there would be only one entry mostly 

 private static final String NON_PARTITIONED = "non-partitioned";

Reply:
add docs for this, related to the reasoning of using cache + key used